pkg/cri/config/config.go

```diff
@@ -416,28 +425,29 @@ func ValidatePluginConfig(ctx context.Context, c *PluginConfig) error {
 			config.Auth = &auth
 			c.Registry.Configs[endpoint] = config
 		}
-		log.G(ctx).Warning("`auths` is deprecated, please use `configs` instead")
+		warnings = append(warnings, deprecation.CRIRegistryAuths)
+		log.G(ctx).Warning("`auths` is deprecated, please use `ImagePullSecrets` instead")
```

Comment:
Would it make sense to leave the logging to the caller of this function? Or would that be complicated, because we also want to log the replacement?

(not a blocker, just thinking out loud)

Reply:
I hadn't considered that.  I think this is easier since the log line message and the deprecation warning message are different, but I don't really have a strong opinion on it.

Reply:
Not a blocker as well; existing code already logs, so it's not a new change in this PR.

Mostly planting "food for thought" here; reduce/avoid logging in library packages where it may not be needed by all consumers of it. That said, containerd's "context" logger can already provide more control over logging to the consumer

(but I just realise that I don't think we have a `WithoutLogger()` utility to disable / remove a logger from a context - perhaps that's something we should think about 🤔)